addons/account/models/chart_template.py

```diff
@@ -333,10 +333,11 @@ def tax_template_changed(tax, template):
                                       and t.type_tax_use == values.get('type_tax_use')\
                                       and t.tax_scope == values.get('tax_scope', False)
                             )
-                        uniq_key = unique_tax_name_key(oldtax)
-                        rename_idx = len(list(filter(lambda t: re.match(fr"^(?:\[old\d*\] |){uniq_key[0]}$", t[0]) and t[1:] == uniq_key[1:], unique_tax_name_keys)))
-                        if rename_idx:
-                            oldtax.name = f"[old{rename_idx - 1 if rename_idx > 1 else ''}] {oldtax.name}"
+                        for index, tax_to_rename in enumerate(oldtax):
+                            uniq_key = unique_tax_name_key(tax_to_rename)
+                            rename_idx = index + len(list(filter(lambda t: re.match(fr"^(?:\[old\d*\] |){uniq_key[0]}$", t[0]) and t[1:] == uniq_key[1:], unique_tax_name_keys)))
+                            if rename_idx:
+                                tax_to_rename.name = f"[old{rename_idx - 1 if rename_idx > 1 else ''}] {tax_to_rename.name}"
```

Comment:
```suggestion
                        uniq_key = unique_tax_name_key(old_tax[0])
                        old_length = len(list(filter(lambda t: re.match(fr"^(?:\[old\d*\] |){uniq_key[0]}$", t[0]) and t[1:] == uniq_key[1:], unique_tax_name_keys)))
                        if old_length:
                            for index, tax_to_rename in enumerate(oldtax):
                                rename_idx = index + old_length
                                tax_to_rename.name = f"[old{rename_idx - 1 if rename_idx > 1 else ''}] {tax_to_rename.name}"
```
Is not it more logical to read this way?  

Normally, there is a Python constraint that should enforce this already.  Not sure how it got past it, but Python constraints for duplicates are not tight anyways.  

Sorry for the late answer.  I had to dive in as well.

Reply:
Thank you for taking the time to check this, it's much appreciated.

IIRC, it was quite easy to bypass the Python constraints. You can edit the name in several languages, but the check is only done in the active session language. So if your UI is in English, you are free to change the name in Spanish to an already existing one. This is how some dbs have repeated names in languages other than English.

Reply:
You mean here: 
![image](https://github.com/user-attachments/assets/cdcfd6ff-7280-4cec-bf1d-a72a002bd9a3)

@filisbits So you are suggesting https://github.com/odoo/odoo/pull/186158 ?

Reply:
Not sure why the EN is on the other side.

Reply:
Exactly. I just tried with a fresh 17 demo db, where I installed es_EC but didn't set it as my language:

```sql
select name->>'en_US',name->>'es_EC',type_tax_use, tax_scope, company_id, country_id  from account_tax where name ->> 'es_EC' ilike '%dupe%'
+----------+-------------------+--------------+-----------+------------+------------+
| ?column? | ?column?          | type_tax_use | tax_scope | company_id | country_id |
|----------+-------------------+--------------+-----------+------------+------------|
| 19% VAT  | 19% IVA dupe test | purchase     | <null>    | 2          | 49         |
| 5% VAT   | 19% IVA dupe test | purchase     | <null>    | 2          | 49         |
+----------+-------------------+--------------+-----------+------------+------------+

```

Reply:
Hi, I've made some changes regarding your suggestion.